fuzzing/private/oss_fuzz/package.bzl

```diff
@@ -29,12 +46,28 @@ def _oss_fuzz_package_impl(ctx):
         outputs = [output_archive],
         inputs = action_inputs,
         command = """
+            set -e
             declare -r STAGING_DIR="$(mktemp --directory -t oss-fuzz-pkg.XXXXXXXXXX)"
             function cleanup() {{
                 rm -rf "$STAGING_DIR"
             }}
             trap cleanup EXIT
-            ln -s "$(pwd)/{binary_path}" "$STAGING_DIR/{base_name}"
+            {binary_runfiles_script}
+            # Create a wrapper that executes the actual fuzz test binary with
+            # the working directory set to what it would be in the Bazel
+            # runfiles tree.
+            # Note: ClusterFuzz may execute fuzz tests with a working directory
+            # different from the containing directory.
+            echo "this_dir=\\$(dirname "\\$0")
+# Allow the Bazel runfiles libraries to find the tree.
+export RUNFILES_DIR=\\$(readlink -f "\\$this_dir/{base_name}.runfiles")
+# LLVMFuzzerTestOneInput - OSS-Fuzz needs this string literal
+# to appear somewhere in the script so it is recognized as a
+# fuzz target.
+cd \\$this_dir/{binary_workdir}
+exec {binary_short_path} \\$@" \
+                > "$STAGING_DIR/{base_name}"
```

Comment:
I think there are two opportunities to simplify this implementation:

1. Instead of creating a shell script of concatenated fragments, you can write the runfiles file mapping to a separate file (we can call it a runfiles manifest), containing pairs of paths on each line. You can use a separate file writing action for this to an intermediate file. Inside the main shell script, you can have a simple while loop that iterates and `read`s each line into a pair of variables and executes the symbolic link creation.
2. Similarly, you can generate the Bash launcher as a separate file using another Bazel action. This way, you can avoid the current Bash-within-Bash situation, which can be confusing for a reader.

WDYT?

Reply:
That sounds good to me. Could you tell me how to write a Bash while loop that moves each line into a pair of variables without any gotchas (spaces in filenames and such things)? I'm not sure I would get this right myself.

Reply:
> That sounds good to me. Could you tell me how to write a Bash while loop that moves each line into a pair of variables without any gotchas (spaces in filenames and such things)? I'm not sure I would get this right myself.

Hmm... very good point about spaces and other special characters :/ My original thought was that we could just have lines of the form `${target} ${link}` and we could use a construct of the form `while read -r target link; do ...; done <{runfiles_manifest}`. But it won't work if there are spaces in the file paths.

However, the ":" character appears to be [invalid in Bazel names](https://docs.bazel.build/versions/master/build-ref.html#name), so perhaps we can use that as a separator? I.e., generate a file where each line is of form `${target}:${link}`, then the shell loop can look like:

```
while IFS=':' read -r target link; do
  # Use here the $target and $link variables.
  mkdir -p ...
  ln -s ...
done <{runfiles_manifest}
```

Reply:
I pushed new commits, the second of which implements the approach using a manifest file and bash while loop. I decided to use the format used by the Bazel runfiles manifest itself. This requires multiple values for `IFS`, but should be relatively gotcha proof.

While I personally slightly prefer the approach of the first commit, I am fine with both.